config.go

```diff
@@ -13,7 +13,7 @@ func saveConfig(pathRelativeToHome string, c Config) error {
 		return fmt.Errorf("finding HOME: %s", err)
 	}
 
-	f, err := os.OpenFile(home+"/"+pathRelativeToHome, os.O_WRONLY|os.O_CREATE, 0600)
+	f, err := os.OpenFile(home+"/"+pathRelativeToHome, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
This was causing issues if we had smaller/shorter tokens or workspace names, ending up with invalid yaml.

Reply:
Not sure what caused the invalid yaml and how `os.O_TRUNC` resolved the issue 😅 Can you expand on that?

Reply:
Yea, so if I have a workspace of 20 characters.. and I switched workspace to one that had 13-14, because we were never truncating the file, the remainder where put at the end of the file after a new line/carriage return.

Under certain conditions (i.e if the remainder of the string contained a space, would become invalid and throw a Marshall failure of the file on the following and all subsequent commands

Reply:
I was able to reproduce the issue after logging in:

```
% clockidup login
error: could not load config: decoding '/home/mvalais/.config/clockidup.yml' from YAML: yaml: line 4: could not find expected ':'
```

As you explained, `~/.config/clockidup.yml` is broken:

```
token: YBABwfrAHCZJzUfi
workspace: aasdasd
-mael-valais
```

What bugs me is that `os.OpenFile` gives a file descriptor where the write pointer is set to the start of the file (i.e., it uses `O_TRUNC` by default). I reproduced this behavior [on the Go Playground](https://play.golang.org/p/QZizSlMuCcn).

It looks like some data race somehow... 🤔


<!-- 

suggestion: let's add a comment about that
```suggestion
	// We write the full configuration file every time, which means we want to overwrite
	// ~/.config/clockidup.yml. By default, os.OpenFile uses O_APPEND on macOS, so we need
	// specify O_TRUNC when opening the file.
	f, err := os.OpenFile(home+"/"+pathRelativeToHome, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
```


-->

Reply:
Aaaahhh I forgot that `O_TRUNC` not only sets the write pointer to the start of the file but also _truncates_ the file, i.e., it sets the byte length to 0.

Before writing the configuration, the file was:

```
token: redacted
workspace: jetstack-mael-valais
```

Without `O_TRUNC`, the length of the file stays the same, which means there are some leftovers from the previous file:

```
token: redacted
workspace: aasdasd
-mael-valais
```

With `O_TRUNC`, the leftover disappears:


```
token: redacted
workspace: aasdasd
```

Reply:
**suggestion:** let's add a comment about that:

```suggestion
	// We re-write the whole configuration file every time we save it. O_TRUNC is thus
	// required to reset the file's length to 0 before writing into it.
	f, err := os.OpenFile(home+"/"+pathRelativeToHome, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
```